Annotato/Annotato/Views/Document/Edit/DocumentView.swift

```diff
@@ -64,38 +76,94 @@ class DocumentView: UIView {
 
     @objc
     private func didChangeVisiblePages(notification: Notification) {
+        showLinkLinesOfVisiblePages()
+        showSelectionBoxedOfVisiblePages()
         showAnnotationsOfVisiblePages()
     }
 
     private func addGestureRecognizers() {
         isUserInteractionEnabled = true
         let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didTap))
         addGestureRecognizer(tapGestureRecognizer)
+
+        let panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(didPan))
+        addGestureRecognizer(panGestureRecognizer)
+    }
+
+    @objc
+    private func didPan(_ sender: UIPanGestureRecognizer) {
+        if sender.state == .began {
+            let startPoint = sender.location(in: self)
+            addSelectionBoxIfWithinBounds(at: startPoint)
+        }
+        if sender.state != .cancelled {
+            let currentTouchPoint = sender.location(in: self)
+            updateCurrentSelectionBoxIfWithinBounds(newEndPointInDocument: currentTouchPoint)
+        }
+        if sender.state == .ended {
+            let currentTouchPoint = sender.location(in: self)
+            updateCurrentSelectionBoxIfWithinBounds(newEndPointInDocument: currentTouchPoint)
+            addAnnotationWithAssociatedSelectionBoxIfWithinBounds()
+        }
     }
 
     @objc
     private func didTap(_ sender: UITapGestureRecognizer) {
         let touchPoint = sender.location(in: self)
         viewModel.setAllAnnotationsOutOfFocus()
-        addAnnotationIfWithinPdfBounds(at: touchPoint)
+        addSelectionBoxIfWithinBounds(at: touchPoint)
+        addAnnotationWithAssociatedSelectionBoxIfWithinBounds()
```

Comment:
```
    @objc
    private func didTap(_ sender: UITapGestureRecognizer) {
        let touchPoint = sender.location(in: self)
        viewModel.setAllAnnotationsOutOfFocus()
        addSelectionBoxIfWithinBounds(at: touchPoint) <--- this one to add the selection box
        addAnnotationWithAssociatedSelectionBoxIfWithinBounds() <--- this one to add the associated annotation
    }
```
I think this can work for the tap because you can add the selection box (with width and height 0) and add the annotation at the same time

But for the didPan, I will need to add the selection box first before the annotation because the view has to display the selection box as the user continues to pan, and only when he releases his finger then the annotation gets added
So I think there needs to be two separate calls